packages/react-reconciler/src/ReactFiberCommitEffects.js

```diff
@@ -244,9 +283,21 @@ export function commitHookEffectListUnmount(
         if ((effect.tag & flags) === flags) {
           // Unmount
           const inst = effect.inst;
+          if (
+            enableUseResourceEffectHook &&
+            effect.kind === ResourceEffectKind &&
+            effect.resource != null &&
+            (effect.create != null ||
+              // TODO(@poteto) this feels gross
+              finishedWork.return == null)
```

Comment:
This only works because in the tests the `App` component is the root component, so there's no `return` fiber to the parent. We need to find a better way to do this that works through multiple layers.

Reply:
I.e. this test fails:

```js
    it('unmounts on deletion after skipped effect', async () => {
      
      function Wrapper(props) {
        return <App {...props} />
      }
      function App({id, username}) {
        const opts = useMemo(() => {
          return {username};
        }, [username]);
        useResourceEffect(
          () => {
            const resource = new Resource(id, opts);
            Scheduler.log(`create(${resource.id}, ${resource.opts.username})`);
            return resource;
          },
          [id],
          resource => {
            resource.update(opts);
            Scheduler.log(`update(${resource.id}, ${resource.opts.username})`);
          },
          [opts],
          resource => {
            resource.destroy();
            Scheduler.log(`destroy(${resource.id}, ${resource.opts.username})`);
          },
        );
        return <Text text={'Id: ' + id} />;
      }

      await act(async () => {
        ReactNoop.render(<Wrapper id={0} username="Sathya" />, () =>
          Scheduler.log('Sync effect'),
        );
        await waitFor(['Id: 0', 'Sync effect']);
        expect(ReactNoop).toMatchRenderedOutput(<span prop="Id: 0" />);
      });

      assertLog(['create(0, Sathya)']);

      await act(async () => {
        ReactNoop.render(<Wrapper id={0} username="Lauren" />, () =>
          Scheduler.log('Sync effect'),
        );
        await waitFor(['Id: 0', 'Sync effect']);
        expect(ReactNoop).toMatchRenderedOutput(<span prop="Id: 0" />);
      });

      assertLog(['update(0, Lauren)']);

      ReactNoop.render(null);
      await waitForAll(['destroy(0, Lauren)']);
      expect(ReactNoop).toMatchRenderedOutput(null);
    });
```